### DISCUSSION TITLE
xcode project

### DISCUSSION THREAD

**Discussion by pis0** [airsdk/Adobe-Runtime-Support#3476] [Ideas]
what about the capabilities of the Air SDK in relation to exporting an Xcode project, like in the android build scenario where we can keep the AndroidStudioProject?

**Comment by ajwfrost** [comment]
Hi

It works in a very different way on iOS: we have a runtime library file which contains all the normal application logic and the VM, including all the start-up code; then your SWF gets compiled into machine code and linked together with this. So there's no actual use of any of the Xcode build tools whilst creating the IPA file (well, the linker I guess, although on Windows we use the LLVM linker).

This would change if we provided AIR more as a library/framework that could be added to render onto a developer's own application, i.e. we just create/render into a UIView. This would mean you could have AIR as part of a wider app, and there could be a 'default' structure/project for creating traditional AIR-only applications. This is something that a few people have asked for, so we might look at this as a low priority task.

I'm just wondering why you are asking though .. is this more for efficiency to avoid long rebuilds, or to help in debugging things, or because you're looking for more flexibility in what you can do with it?

thanks

**Reply by pis0** [reply]
thanks for the quick response! @ajwfrost 

we're looking for more flexibility, for sure... as a company that develops/maintains 
a lot of small games (mostly made in as3), the ability to incorporate these games 
in isolation, into a standard main project like you said, would be really nice.. 
not only technically but also from a business point of view.

glad you're looking into this! 
even if it's low priority.
